packages/graphql-yoga/src/server.ts

```diff
@@ -226,20 +226,35 @@ export class YogaServer<
             }
         : logger
 
+    const maskErrorFn =
+      (typeof options?.maskedErrors === 'object' &&
+        options.maskedErrors.maskError) ||
+      yogaDefaultFormatError
+
     this.maskedErrorsOpts =
       options?.maskedErrors === false
         ? null
         : {
-            maskError: (error, message) =>
-              yogaDefaultFormatError({
-                error,
-                message,
-                isDev: this.maskedErrorsOpts?.isDev ?? false,
-              }),
             errorMessage: 'Unexpected error.',
             ...(typeof options?.maskedErrors === 'object'
               ? options.maskedErrors
               : {}),
+            maskError: (error, message) => {
+              const newError = maskErrorFn(
+                {
+                  error,
+                  message,
+                  isDev: this.maskedErrorsOpts?.isDev ?? false,
+                },
+                message,
+              )
+
+              if (newError !== error) {
+                this.logger.error(error)
+              }
+
+              return newError
+            },
```

Comment:
In case error masking is disabled - thus every error is an expected error - thus we shouldn't log any error.